service/src/main/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataValidatorService.java

```diff
@@ -0,0 +1,42 @@
+package uk.gov.hmcts.reform.fpl.service.robotics;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.fpl.exceptions.robotics.RoboticsDataException;
+import uk.gov.hmcts.reform.fpl.model.robotics.RoboticsData;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.validation.ConstraintViolation;
+import javax.validation.Validator;
+
+import static java.lang.String.format;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class RoboticsDataValidatorService {
+    private final Validator validator;
+
+    public List<String> validationErrors(final RoboticsData roboticsData) {
+        Set<ConstraintViolation<RoboticsData>> violations = validator.validate(roboticsData);
+
+        return violations.stream().filter(Objects::nonNull)
+            .map(this::formatViolationMessage)
+            .collect(Collectors.toList());
+    }
+
+    public void verifyRoboticsJsonData(final String roboticsJsonData) {
```

Comment:
minor, these 2 methods are a bit inconsistent, one throws exception when validation fails, second let client decide what to do